app/src/ui/branches/branches-container.tsx

```diff
@@ -208,13 +203,32 @@ export class BranchesContainer extends React.Component<
         onItemClick={this.onPullRequestClicked}
         onDismiss={this.onDismiss}
         renderPostFilter={this.renderPullRequestPostFilter}
+        renderListHeader={this.renderPullRequestsHeader}
         dispatcher={this.props.dispatcher}
         repository={repository}
         isLoadingPullRequests={this.props.isLoadingPullRequests}
       />
     )
   }
 
+  private getRepositoryName(): string {
+    const gitHubRepository = this.getGitHubRepository()
+
+    if (gitHubRepository === null) {
+      return this.props.repository.name
+    }
+
+    return gitHubRepository.fullName
+  }
```

Comment:
@rafeca is the primary purpose of this to fallback to the plain repository name if there's no GitHubRepository for it? how does this differ from `nameOf`, otherwise?

Reply:
The main difference with `nameOf` is that this function tries to use the parent repository if it exists (it's unfortunate that I couldn't use `nameOf`).

But now that I think of this, this function is only used for the pull requests view. This means that I can it to receive a `RepositoryWithGitHubRepository` argument instead of using the `repository` from the props and then I won't need the fallback for non-github repositories.

Reply:
I've ended up changing the strategy a little bit: Instead of start using a `repository` argument on each method that has the more specific type than `this.props.repository` (this felt like a code smell), I've ended up passing the `repository` object to the `<PullRequestList/>` component and moved all the pullrequest-specific logic to that component so it can freely use `this.props.repository` knowing that the object is a `RepositoryWithGitHubRepository`.

This has the added benefit of compartmentalizing a little bit more the logic (logic specific to the `<PullRequestList/>` list doesn't leak to `<BranchContainer/>`). We can do the same for `<BranchList/>` in a separate PR.

This has been done in e2597febe5b0116e9e58ddd692dbeb3285f3ea4a. If you think that the commit is out of the scope of this PR I can move it to a separate one.